src/utils.jl

```diff
@@ -77,6 +77,51 @@ function sideof(p::Point{2,T}, c::Chain{2,T}) where {T}
   ifelse(isapprox(w, zero(T), atol=atol(T)), :OUTSIDE, :INSIDE)
 end
 
+"""
+    sideof(point, mesh)
+
+Determines whether a `point` is inside, outside or on the surface of a `mesh`.
+Possible results are `:INSIDE`, `:OUTSIDE`, or `:ON`.
+
+## Notes
+
+Uses a ray-casting algorithm.
+"""
+function sideof(point::Point{3,T}, mesh::Mesh{3,T}) where {T}
+  if !(eltype(mesh) <: Triangle)
+    throw(ArgumentError(
+          "This function only works for surface meshes with triangles as elements."))
+  end
+
+  z = last.(coordinates.(extrema(mesh)))
+  r = Ray(point, Vec(zero(T), zero(T), 2 * (z[2] - z[1])))
+  vs = vertices(mesh)
+
+  intersects = false
+  edgecrosses = 0
+  cornerscrossing = Point{3,T}[]
+  for elem in mesh
+    I = intersection(r, elem)
```

Comment:
Should we use the 3-arg version of intersection to throw away other branches at compile time? Check the docs of intersection for examples.

Reply:
I don't think it would be able to since the compiler wouldn't know the type of the intersection until we actually run the intersection so I think both methods would be equivalent performance wise. I will update it for style reasons, however.

Reply:
I think the 3-arg version is only beneficial in performance when you have a mesh with different types of elements (like triangles and quads, etc) but since we limit this to triangles only (for now) then it shouldn't make a difference.

Reply:
I don't think this is related to multiple element types. The 3-arg version simply sends the relevant branches that are explicitly written to a new function barrier. So it can generate typed code instead of Any returns.